### PR TITLE
Fix -Wunused-parameter, -Wmissing-field-initializers

### PR DESCRIPTION
**Pull request by rurban**
Working towards proper --strict-warnings.
Also add some field name comments.
Replace -Qunused-arguments with -Wno-error=unused-command-line-argument.
    
Tested with ./Configure CC=gcc|clang --strict-warnings enable-asan \
      enable-ubsan enable-fips enable-quic enable-rc5 enable-ssl3 \
      enable-ssl3-method enable-tfo enable-trace enable-unit-test enable-uplink \
      enable-weak-ssl-ciphers enable-zlib enable-zlib-dynamic enable-zstd \
      enable-zstd-dynamic && m test
and the defaults.

CLA on its way

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

File: apps/asn1parse.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ const OPTIONS asn1parse_options[] = {
     {"dump", OPT_DUMP, 0, "unknown data in hex form"},
     {"dlimit", OPT_DLIMIT, 'p',
      "dump the first arg bytes of unknown data in hex form"},
-    {NULL}
+    {NULL, 0, 0, NULL}
```
Comment (t8m) [inline apps/asn1parse.c:56]: I do not think we want this kind of changes.

File: apps/ciphers.c [hunk 2]
```diff
@@ -81,6 +81,12 @@ static unsigned int dummy_psk(SSL *ssl, const char *hint, char *identity,
                               unsigned char *psk,
                               unsigned int max_psk_len)
 {
+    (void)ssl;
```
Comment (t8m) [inline apps/ciphers.c:84]: Neither these.

**Review by t8m — COMMENTED** [review]
IMO OTC should decide if we want such kind of changes to appease these warnings

File: apps/ciphers.c (hunk 2 above)
Comment (thesamesam) [reply apps/ciphers.c:84]: I think this one is too noisy, agreed.

File: apps/asn1parse.c (hunk 1 above)
Comment (thesamesam) [reply apps/asn1parse.c:56]: IMO this one is maybe worth having to be able to enable the error.

File: apps/ciphers.c (hunk 2 above)
Comment (rurban) [reply apps/ciphers.c:84]: These are for -Wunused-parameter. There are thousands of such silenced unused params, which are hereby at least documented.

Comment (thesamesam) [reply apps/ciphers.c:84]: I'm saying that silencing it is not particularly useful here -- you're making it harder to then discover where you want actual refactoring to drop the unused parameters. This is like turning off the smoke detector to drown out the noise.

We go from "here's a nice list of things where there's unused parameters we should fix" to "the warning is clean but we know we have tonnes of suppressions everywhere which are hard to grep for".

You'd be better off passing -Wno-unused-parameter, as you're not then hiding the problem.

Comment (rurban) [reply apps/ciphers.c:84]: You got it reverse. 
* With  -Wno-unused-parameter you are silencing all such errors, and have no idea which params are ignored.
* With (void)param you enable the warnings, detect new unused params and if the param is used later, you get an error also.

**Comment by hlandau** [comment]
Personally I do not think these warnings are valuable. I do not think we want to start adding `(void)foo` to every function where a parameter is unused, or start having to explicitly list every entry in an array. The latter is especially troublesome if the array is quite large (e.g. if it is a 4k buffer which intentionally contains data only at the beginning).

**Comment by rurban** [comment]
The CLA is now in your database.

**Review by FdaSilvaYY — COMMENTED** [review]
This PR is too large to be reviewed and merged.
Majority of the changes have no value added :
In first, changes related to unused arguments.
the missing methods comments and scripts changes are valuable.

**Comment by t8m** [comment]
**OTC decided that we do not want to change the code to placate these warnings for now.**

**Comment by t8m** [comment]
I am closing this PR.

**Pull request closed without merging**